src/next/mcp-api-handler.ts

```diff
@@ -345,6 +345,27 @@ export function initializeMcpApiHandler(
         }
       }
     } else if (url.pathname === sseEndpoint) {
+      // Check HTTP method - only allow GET for SSE connections
+      if (req.method !== "GET") {
+        logger.log(`Rejected SSE connection with method ${req.method}`);
+        res
+          .writeHead(405, { "Content-Type": "text/plain" })
+          .end("Method Not Allowed");
+        return;
+      }
+
+      // Check that Accept header supports event-stream
+      const acceptHeader = req.headers.get("accept");
+      if (acceptHeader && !acceptHeader.includes("text/event-stream")) {
```

Comment:
This could false positive for `Accept: */*` or `Accept: text/*`. Options I see:
- to use a library like [negotiator](https://www.npmjs.com/package/negotiator)
- simply check for those two cases as well
- not enforce the `Accept` header, since the method check is probably enough
either of these is fine I think

Reply:
hmm can you elaborate on it being a false positive when `Accept: */*` or `Accept: text/*` ?

if it's `Accept: text/json` it'll be true && true and not allowed

it **needs** to be text/event-stream to pass this?